core/store/memory_directory.cpp

```diff
@@ -538,19 +538,22 @@ bool memory_directory::rename(
   try {
     auto lock = make_lock_guard(mutex);
 
-    // prevent rehashing so we can re-use 'it' to erase an element
-    files_.reserve(files_.size() + 1);
+    auto res = files_.emplace(
+      std::piecewise_construct,
+      std::forward_as_tuple(dst),
+      std::forward_as_tuple());
 
-    const auto it = files_.find(src);
+    if (IRS_LIKELY(res.second)) {
+      auto it = files_.find(src);
 
-    if (it == files_.end()) {
-      return false;
+      if (IRS_LIKELY(it != files_.end())) {
+        res.first->second = std::move(it->second);
+        files_.erase(it);
+        return true;
+      }
     }
 
-    files_[dst] = std::move(it->second);
-    files_.erase(it);
-
-    return true;
+    files_.erase(res.first);
```

Comment:
if `dst` already present (or `src`  is not found) in `files_` this line will remove `dst` (and leave `src` as is). Is that intentional? Old version was overwriting `dst` in the case of existing `dst` and leaving `dst` as is in case of  `src` was not found .